sympy/functions/special/gamma_functions.py

```diff
@@ -319,6 +319,8 @@ def eval(cls, a, x):
 
         # Special values.
         if a.is_Number:
+            if a is S.Zero and x.is_zero is False:
```

Comment:
a = 0 is not the only pole of lowergamma. They should preferably all be handled in the same place.

Reply:
Yeah correct, I was reading wiki just now to make sure I'm not missing something when I realized this and as you pointed the check needs improvement , I am also doubtful about few other cases . Will makes the changes as I get more clarity here !